cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowFilterForButtonFiltersCorrectField(
+  docTableRowNumber,
+  expandedDocumentRowNumber,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied
+) {
+  getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber).then(
+    ($expandedDocumentRowValue) => {
+      const filterFieldText = $expandedDocumentRowValue.text();
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('addInclusiveFilterButton').click();
+      });
+      // Verify pill text
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
+      }).should('have.text', filterFieldText);
+      cy.getElementByTestId('discoverQueryHits').should(
+        'have.text',
+        expectedQueryHitsAfterFilterApplied
+      ); // checkQueryHitText must be in front of checking first line text to give time for DocTable to update.
+      getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber).should(
+        'have.text',
+        filterFieldText
+      );
+    }
+  );
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Check the Filter Out button in the expandedDocumentRowNumberth field in the expanded Document filters the correct value.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowFilterOutButtonFiltersCorrectField(
+  docTableRowNumber,
+  expandedDocumentRowNumber,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied
+) {
+  getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber).then(
+    ($expandedDocumentRowValue) => {
+      const filterFieldText = $expandedDocumentRowValue.text();
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('removeInclusiveFilterButton').click();
+      });
+      // Verify pill text
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
```

Comment:
we have a `default` global timeout set to a ridiculous number. Did that not get applied for you?

Reply:
Updated to remove these timeouts.